hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -496,18 +504,15 @@ def test_function(self, data: ConjectureData) -> None:
 
         if data.status == Status.INTERESTING:
             if self.settings.backend != "hypothesis":
-                for node in data.examples.ir_tree_nodes:
-                    value = data.provider.post_test_case_hook(node.value)
-                    # require providers to return something valid here.
-                    assert (
-                        value is not None
-                    ), "providers must return a non-null value from post_test_case_hook"
-                    node.value = value
-
                 # drive the ir tree through the test function to convert it
                 # to a buffer
                 data = ConjectureData.for_ir_tree(data.examples.ir_tree_nodes)
                 self.__stoppable_test_function(data)
+                data.freeze()
+                # should we raise Flaky here instead?
+                if data.status != Status.INTERESTING:
+                    self.exit_with(ExitReason.flaky)
```

Comment:
I think exit-flaky is correct, but we'll also need a covering test 🙂

Reply:
just exiting is a silent conclusion of the run, so after looking at it I think we do not want this. I put a stop-gap flaky raise but we lose eg the nice custom deadline reporting we get at the `StateForActualGivenExecution` level. I don't see an easy way to thread things through at this lower level but I am also having a hard time keeping the engine control flow graph straight. I think this is acceptable for now.